### PR TITLE
fix(data-table-v2): fix expandable cell index

### PR DESCRIPTION
**Pull request by asudoh**
v10 markup adds `bx--table-expand-v2` class on `<th>` in addition to `<td>` presumably to better align checkbox next to expando (below style rule), but JS code had assumed that that class applied only on `<td>`, causing wrong index in an array to determine where expandable row content should be put to.

```css
th.bx--table-expand-v2 + th.bx--table-column-checkbox {
  padding-left: 0.5rem;
}
```

This was discovered by running unit tests on `v10` markup.

Refs #2209.

#### Changelog

**Changed**

- The CSS selector to find expando cells in data table, coping with `v10` markup change.

#### Testing / Reviewing

Testing should make sure our @jnm2377's data table work is not broken, especially on expandable row support.

### CODE REVIEW

**Comment by asudoh** [comment]
@shinytoyrobots @alisonjoseph One line change to data table code - OK to put this to `v10`? Thanks!

**Review by jnm2377 — APPROVED** [review]
Ah I see what you were referring to. Looks good! 👍

**Review by emyarod — APPROVED** [review]
looks good to me! waiting for milestone change approval before merging

**Review by joshblack — APPROVED** [review]

**Pull request merged**
